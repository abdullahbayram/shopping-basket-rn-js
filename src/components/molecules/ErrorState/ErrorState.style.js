import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  container: {
    alignItems: 'center',
    justifyContent: 'center',
    padding: 16,
  },
  errorText: {
    marginBottom: 8,
    textAlign: 'center',
  },
});
