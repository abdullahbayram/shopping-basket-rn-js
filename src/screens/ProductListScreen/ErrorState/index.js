import React from 'react';
import { View } from 'react-native';
import { Button, HelperText } from '@components/atoms';
import PropTypes from 'prop-types';
import styles from './ErrorState.style';

const ErrorState = ({ errorMessage, onRetry }) => {
  return (
    <View style={styles.container}>
      <HelperText style={styles.errorText} type="error">
        {errorMessage}
      </HelperText>
      <Button onPress={onRetry} mode="contained">
        Retry
      </Button>
    </View>
  );
};

ErrorState.propTypes = {
  errorMessage: PropTypes.string.isRequired,
  onRetry: PropTypes.func.isRequired,
};

export default ErrorState;
