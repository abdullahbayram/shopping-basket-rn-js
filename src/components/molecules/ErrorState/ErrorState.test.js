import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react-native';
import ErrorState from '.';

describe('ErrorState Component', () => {
  const mockRetry = jest.fn();

  it('renders the error message', () => {
    render(<ErrorState errorMessage="Test error message" onRetry={mockRetry} />);
    expect(screen.getByText('Test error message')).toBeTruthy();
  });

  it('calls the retry function when the Retry button is pressed', () => {
    render(<ErrorState errorMessage="Another error" onRetry={mockRetry} />);

    const retryButton = screen.getByText('Retry');
    fireEvent.press(retryButton);

    expect(mockRetry).toHaveBeenCalledTimes(1);
  });

  it('matches the snapshot', () => {
    const { toJSON } = render(<ErrorState errorMessage="Snapshot error" onRetry={mockRetry} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
