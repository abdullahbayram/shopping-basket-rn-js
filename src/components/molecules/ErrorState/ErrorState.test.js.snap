// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorState Component matches the snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "justifyContent": "center",
      "padding": 16,
    }
  }
>
  <Text
    collapsable={false}
    maxFontSizeMultiplier={1.5}
    onLayout={[Function]}
    style={
      {
        "color": "rgba(179, 38, 30, 1)",
        "fontFamily": "System",
        "fontSize": 12,
        "fontWeight": "400",
        "letterSpacing": 0.25,
        "lineHeight": 20,
        "marginBottom": 8,
        "opacity": 1,
        "paddingHorizontal": 12,
        "paddingVertical": 0,
        "textAlign": "center",
        "transform": [
          {
            "translateY": -0,
          },
        ],
        "writingDirection": "ltr",
      }
    }
  >
    Snapshot error
  </Text>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "rgba(98, 91, 113, 1)",
        "borderRadius": 20,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
      }
    }
    testID="button-container-outer-layer"
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "rgba(98, 91, 113, 1)",
          "borderColor": "transparent",
          "borderRadius": 20,
          "borderStyle": "solid",
          "borderWidth": 0,
          "flex": undefined,
          "minWidth": 64,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
      testID="button-container"
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "borderRadius": 20,
            },
          ]
        }
        testID="button"
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <Text
            numberOfLines={1}
            selectable={false}
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "letterSpacing": 0.1,
                    "lineHeight": 20,
                  },
                  [
                    {
                      "marginHorizontal": 16,
                      "marginVertical": 9,
                      "textAlign": "center",
                    },
                    false,
                    {
                      "marginHorizontal": 24,
                      "marginVertical": 10,
                    },
                    undefined,
                    false,
                    {
                      "color": "rgba(125, 82, 96, 1)",
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    undefined,
                  ],
                ],
              ]
            }
            testID="button-text"
          >
            Retry
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
